### PR TITLE
Make statistics clickable

### LINKED ISSUE #248: Make statistics clickable
# Feature request

I'd like to request, that when clicking on one of the 4 buttons (Farming, Online, Offline, Disabled) one should be redirected to bots page filtering it by the button one clicked on. So if i click on Farming i only want to see bots that are currently farming

image

### PR DESCRIPTION
**Pull request by MrBurrBurr**
Closes #248

### CODE REVIEW

**Comment by JustArchi** [comment]
> fuck you codeclimate

Exactly why I don't use such things in my own projects, they usually do more harm than good, returns there were far more readable than the current flow, but I'm not in charge of ASF-ui 😀

**Comment by Aareksio** [comment]
Revert `fuck you codeclimate` please. I disabled `too many returns` warnings.

File: src/components/NavigationBots.vue [hunk 1]
```diff
@@ -39,6 +40,21 @@
 			},
 			onTransitionEnd() {
 				this.transitioning = false;
+			},
+			select(type) {
+				if (this.$route.name !== 'bots') this.$router.replace({ name: 'bots' });
```
Comment (Aareksio) [inline src/components/NavigationBots.vue]:
```suggestion
				if (this.$route.name !== 'bots') this.$router.push({ name: 'bots' });
```

File: src/components/NavigationBots.vue [hunk 2]
```diff
@@ -16,6 +16,7 @@
 		components: { FitText },
 		data() {
 			return {
+				selectedBots: this.$store.getters['settings/selectedBots'],
```
Comment (Aareksio) [inline src/components/NavigationBots.vue]: Make this a computed property and use vuex actions to mutate it's state.
OR
Make it local variable, skip the store completly. We can use query parameter for this - `?filter=online`

**Pull request merged**